lib/Conversion/TorchToLinalg/DataMovement.cpp

```diff
@@ -789,6 +792,81 @@ class ConvertAtenViewOp : public OpConversionPattern<AtenViewOp> {
     return success();
   }
 
+  // Starting from the beginning of the dims arrays, this helper finds the
+  // smallest set of consecutive dims in each array that satisfies one of
+  // the following conditions.
+  // 1. The product of the static dim sizes in the two subsets is equal.
+  // 2. The product of the dim size multiplied by the multiplier for the unknown
+  // one in both subsets is equal.
+  // The indices arrays are populated with the indices of the dims arrays that
+  // correspond to the subsets found.
+  //
+  // An error is returned if two subsets of dims with total number of elements
+  // equal to each other is not found.
+  static LogicalResult mapParallelUnknownDims(ArrayRef<int64_t> xDims,
+                                              ArrayRef<int64_t> yDims,
+                                              SmallVector<int64_t> &xIndices,
+                                              SmallVector<int64_t> &yIndices,
+                                              int64_t xMultiplier,
+                                              int64_t yMultiplier) {
+    if (xDims.empty() || yDims.empty())
+      return failure();
+    if (llvm::count(xDims, kUnknownSize) > 1 ||
+        llvm::count(yDims, kUnknownSize) > 1)
+      return failure();
+
+    int64_t xTotalSize = xDims[0];
+    int64_t yTotalSize = yDims[0];
+    SmallVector<int64_t> xIndicesResult({0});
+    SmallVector<int64_t> yIndicesResult({0});
+    size_t nextXIndex = 1;
+    size_t nextYIndex = 1;
+    bool xHasUnknownSize = false;
+    bool yHasUnknownSize = false;
```

Comment:
nit: there is no need for these, since it is already ensured by the `llvm::count` check above that there is at most one unknown size in each array

Reply:
It is necessary. because the bool flags updated in the while loop.

Reply:
Makes sense, thanks!